stl/inc/xstring

```diff
@@ -3026,27 +3187,34 @@ public:
 
 #if _HAS_CXX17
     template <class _StringViewIsh, _Is_string_view_ish<_StringViewIsh> = 0>
-    basic_string& assign(const _StringViewIsh& _Right) {
+    _CONSTEXPR20_DYNALLOC basic_string& assign(const _StringViewIsh& _Right) {
         const basic_string_view<_Elem, _Traits> _As_view = _Right;
         return assign(_As_view.data(), _Convert_size<size_type>(_As_view.size()));
     }
 
     template <class _StringViewIsh, _Is_string_view_ish<_StringViewIsh> = 0>
-    basic_string& assign(const _StringViewIsh& _Right, const size_type _Roff, const size_type _Count = npos) {
+    _CONSTEXPR20_DYNALLOC basic_string& assign(
+        const _StringViewIsh& _Right, const size_type _Roff, const size_type _Count = npos) {
         // assign _Right [_Roff, _Roff + _Count)
         basic_string_view<_Elem, _Traits> _As_view = _Right;
         return assign(_As_view.substr(_Roff, _Count));
     }
 #endif // _HAS_CXX17
 
-    basic_string& assign(_In_reads_(_Count) const _Elem* const _Ptr, _CRT_GUARDOVERFLOW const size_type _Count) {
+    _CONSTEXPR20_DYNALLOC basic_string& assign(
+        _In_reads_(_Count) const _Elem* const _Ptr, _CRT_GUARDOVERFLOW const size_type _Count) {
         // assign [_Ptr, _Ptr + _Count)
-        if (_Count <= _Mypair._Myval2._Myres) {
-            _Elem* const _Old_ptr   = _Mypair._Myval2._Myptr();
-            _Mypair._Myval2._Mysize = _Count;
-            _Traits::move(_Old_ptr, _Ptr, _Count);
-            _Traits::assign(_Old_ptr[_Count], _Elem());
-            return *this;
+#ifdef __cpp_lib_is_constant_evaluated
+        if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_is_constant_evaluated
+        {
+            if (_Count <= _Mypair._Myval2._Myres) {
+                _Elem* const _Old_ptr   = _Mypair._Myval2._Myptr();
+                _Mypair._Myval2._Mysize = _Count;
+                _Traits::move(_Old_ptr, _Ptr, _Count);
+                _Traits::assign(_Old_ptr[_Count], _Elem());
+                return *this;
+            }

```

Comment:
Performance note: I believe this is not the best solution. 

We should only need this if capacity of the string is equal to the minimal capacity of `_BUFF_SIZE` because otherwise we would already have plastered this with data